File: flippinganimage.js
//Objective is to flip an image - reverse every row then invert every tile.

let matrix = 
[[1,0,0],
 [0,1,0],
 [0,0,1]]


//O(n) solution that reverses each row than inverts each tile.

//Reverse each row
for (let i = 0; i < matrix.length; i++) {
    matrix[i] = matrix[i].reverse()
}

//Invert each tile
for (let i = 0; i < matrix.length; i++) {
    for (let j = 0; j < matrix[0].length; j++) {
        matrix[i][j] = !matrix[i][j]
    }
}

return matrix

File: flippinganimage.java
//Java Solution

class Solution {
    public int[][] flipAndInvertImage(int[][] A) {
        for (int[] row : A) {
            row = reverse(row);
        }
        
        for (int i = 0; i < A.length; i++) {
            for (int j = 0; j < A[i].length; j++) {
                A[i][j] = A[i][j] == 1 ? 0 : 1;
            }
        }
        
        return A;
    }
    
    public int[] reverse(int[] row) {
        int left = 0;
        int right = row.length - 1;
        while (left < right) {
            int temp = row[left];
            row[left++] = row[right];
            row[right--] = temp;
        }
        
        return row;
    }
}